src/NuGet.Clients/NuGet.PackageManagement.UI/Options/ConfigurationFilesViewModel.cs

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Windows.Input;
+using Microsoft.VisualStudio.ComponentModelHost;
+using Microsoft.VisualStudio.PlatformUI;
+using Microsoft.VisualStudio.Services.Common;
+using NuGet.ProjectManagement;
+using NuGet.VisualStudio;
+
+namespace NuGet.PackageManagement.UI.Options
+{
+    public class ConfigurationFilesViewModel
+    {
+        public ObservableCollection<string> ConfigurationFilesCollection { get; private set; }
+        public string SelectedPath { get; set; }
+        public ICommand OpenConfigurationFile { get; set; }
+
+        public ConfigurationFilesViewModel()
+        {
+            ConfigurationFilesCollection = new ObservableCollection<string>();
+            OpenConfigurationFile = new DelegateCommand(ExecuteOpenConfigurationFile, IsSelectedPath, NuGetUIThreadHelper.JoinableTaskFactory);
+        }
+
+        private bool IsSelectedPath()
+        {
+            return SelectedPath != null;
+        }
+
+        private void ExecuteOpenConfigurationFile()
+        {
+            OpenConfigFile(SelectedPath);
+        }
+        internal void InitializeOnActivated(CancellationToken cancellationToken)
+        {
+            SetConfigPaths();
+        }
+
+        public void SetConfigPaths()
+        {
+            IComponentModel componentModelMapping = NuGetUIThreadHelper.JoinableTaskFactory.Run(ServiceLocator.GetComponentModelAsync);
+            var settings = componentModelMapping.GetService<Configuration.ISettings>();
+            IReadOnlyList<string> configPaths = settings.GetConfigFilePaths().ToList();
+            ConfigurationFilesCollection.Clear();
+            ConfigurationFilesCollection.AddRange(configPaths);
+        }
+
+        private void OpenConfigFile(string selectedPath)
+        {
+            var componentModel = NuGetUIThreadHelper.JoinableTaskFactory.Run(ServiceLocator.GetComponentModelAsync);
+            var projectContext = componentModel.GetService<INuGetProjectContext>();
+
+            // This check is performed in case the user moves or deletes a config file while they have it selected in the Options window.
+            if (!File.Exists(selectedPath))
+            {
+                var error = new FileNotFoundException(selectedPath);
+                MessageHelper.ShowErrorMessage(error.Message, Resources.ShowError_FileNotFound);
```

Comment:
What does this actually show? Did you intend to pass the exception, or the message? 
The `FileNotFoundException.Message` default message appears to be very similar to `Resources.ShowError_FileNotFound`. Do we need both? Is there a screenshot of how this error looks?

```suggestion
                MessageHelper.ShowErrorMessage(Resources.ShowError_FileNotFound);
```

Reply:
Here's a screenshot of what the error message looks like:
<img alt="Options Error Message" src="https://github.com/NuGet/NuGet.Client/assets/85927389/9fdbbe4d-ad86-48e4-b82d-ef940e351492">
`error.Message` is the file path, and `Resources.ShowError_FileNotFound` is "The following file could not be found:".

Reply:
How about doing a string.Format or something similar to avoid creating new instance of `FileNotFoundException` just to obtain the error message?

string.Format(Resources.ShowError_FileNotFound, selectedPath);